utils/exceptions.cc

```diff
@@ -63,3 +66,18 @@ bool should_stop_on_system_error(const std::system_error& e) {
     }
     return true;
 }
+
+std::atomic<bool> abort_on_internal_error{false};
+
+void set_abort_on_internal_error(bool do_abort) {
+    abort_on_internal_error.store(do_abort);
+}
+
+void on_internal_error(seastar::logger& logger, const seastar::sstring& msg) {
+    if (abort_on_internal_error.load()) {
+        logger.error(msg.c_str());
+        abort();
+    } else {
+        throw std::runtime_error(msg.c_str());
```

Comment:
I think it would be helpful to print the backtrace here to help identify the code path.

Reply:
Yes.